### PR TITLE
add deno.pathBackwards and deno.pathForwards

### PR DESCRIPTION
**Pull request by ztplz**
Add convert tools that described in #957

### CODE REVIEW

File: js/path.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+// Copyright 2018 the Deno authors. All rights reserved. MIT license.
+
+export function pathBackwards(path: string): string {
+  path.trim();
```
Comment (demurgos) [inline js/path.ts]: `trim` returns a new value, it should be `path = path.trim()` (same issue in the other function).
There may also be edge cases around trimming, it would benefit from having test cases checking it.

Comment (ztplz) [reply js/path.ts]: Thanks,  fixed it.

File: js/path.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  path = path.replace(/\//g, "\\");
+  const double = /\\\\/;
+  while (double.test(path)) {
+    path = path.replace(double, "\\");
+  }
+  return path;
+}
+
+export function pathForwards(path: string): string {
+  path = path.trim();
```
Comment (kitsonk) [inline js/path.ts]: I really hate reassignment to arguments, because especially when it is happens with non-primatives, as it is just totally confusing.  Also, the reassignments are a bit confusing.  I would think this would be a lot cleaner:

```ts
let result = path
  .trim()
  .replace(/^([a-zA-Z]+:|\.\/)/, "")
  .replace(/\\/g, "/");
const double = /\/\//;
while (double.test(result)) {
  result = result.replace(result, "/");
}
return result;
```

Also, is there a specific edge case that would not cause the following to work:

```ts
return path
  .trim()
  .replace(/^([a-zA-Z]+:|\.\/)/, "")
  .replace(/\\/g, "/");
  .replace(/\/{2}/g, "/");
```

File: js/path.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+// Copyright 2018 the Deno authors. All rights reserved. MIT license.
+
+export function pathBackwards(path: string): string {
+  path = path.trim();
```
Comment (kitsonk) [inline js/path.ts]: Same thoughts as below for `pathForwards`

**Comment by hayd** [comment]
I know the naming is from #957, but it feels a little awkward. Would this be better exposed as a `Path` type with these as constructor/to string methods?

Then using `Path` internally might make it easier to get the desired outcome:

> If we ensure that all APIs in Deno return forward slashes, we can alleviate this cross-platform issue from our users.

**Comment by ry** [comment]
@piscisaureus Please review.

**Comment by piscisaureus** [comment]
I don't believe these APIs serve any purpose. One might think that if you change all backward slashes to forward slashes the problem of "windows paths are ridiculous" just goes away, but that's naive.

* You're still gonna encounter windows-y paths:
  * In environment variables, e.g. `PATH`.
  * In command-line arguments.
  * In files that that reference other files.
* You're still going to need to produce windows-y paths:
  * To set environment variables where forward slashes aren't accepted (e.g. PATH).
  * To pass as arguments to subprocesses that don't accept them (note that traditionally on windows `/` is how flags are set; e.g. the equivalent of `rm -rf /foo` is `rd /s/q \foo`).
* Path semantics -- which is where usually the real pain lies -- is still different:
  * `c:/` is a root.
  * `c:/../` equals `c:/`
  * When resolving symlinks, `..` is eagerly applied vs lazily on unix.
  * `/foo/bar/baz` is a relative path on windows.

However, these APIs do create new problems of their own:
* At least on linux, a valid filename can start and/or end with whitespace. If you trim it, you're changing the path. That's maybe not a big deal when you're creating a new file, but it is a problem when you're trying to open an existing file.
* A backslash is a character that is valid in a unix path. The `pathBackwards` transformation is never really correct on unix.
* Merging consecutive slashes -> this is not correct on windows, because consecutive slashes can be significant, e.g. at the start of a path to distinguish between root-relative (`\foo\bar`), UNC paths (`\\myserver\share\foo\bar`) and device paths (`\\.\pipe\my-pipe`).

If there's an any opportunity for beating some reason into windows I'll be the first to do it, and there's definitely things we can do about paths. This however isn't the way to go about it.
